modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/SalesforceObjectEntryManagerImpl.java

```diff
@@ -273,20 +278,68 @@ public ObjectEntry updateObjectEntry(
 		return null;
 	}
 
+	private List<String> _getAccountReferenceCodesByUserId(long userId)
+		throws Exception {
+
+		List<String> accountExternalReferenceCodes = new ArrayList<>();
+
+		for (AccountEntryUserRel accountEntryUserRel :
+				_accountEntryUserRelLocalService.
+					getAccountEntryUserRelsByAccountUserId(userId)) {
+
+			AccountEntry accountEntry = accountEntryUserRel.getAccountEntry();
+
+			accountExternalReferenceCodes.add(
+				accountEntry.getExternalReferenceCode());
+		}
+
+		return accountExternalReferenceCodes;
+	}
+
 	private DateFormat _getDateFormat() {
 		return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
 	}
 
+	private String _getFilters(
+			long companyId, DTOConverterContext dtoConverterContext,
+			ObjectDefinition objectDefinition, String scopeKey)
+		throws Exception {
+
+		String filterString = null;
+
+		if (_isPermissionEnabled(companyId, objectDefinition, scopeKey) &&
+			objectDefinition.isAccountEntryRestricted()) {
+
+			ObjectField objectField = _objectFieldLocalService.getObjectField(
+				objectDefinition.getAccountEntryRestrictedObjectFieldId());
+
+			filterString = StringBundler.concat(
+				objectField.getExternalReferenceCode(), " IN ('",
+				StringUtil.merge(
+					_getAccountReferenceCodesByUserId(
+						dtoConverterContext.getUserId()),
+					", '"),
+				"')");
+		}
+
+		if (filterString == null) {
+			return StringPool.BLANK;
+		}
```

Comment:
Put the first if conditional in reverse order like: 

if( !_isPermissionEnable OR !objectDefinition.isAccountEntryRestricted() ) {
return StringPool.Blank;
}

And than you'll have 1 less if clause